draft-kwbdgrr-tsvwg-net-collab-rqmts.md

```diff
@@ -91,32 +93,66 @@ For example, live media or AI generated content can have significant dynamic var
+When a bottleneck exists temporarily, the network has no choice but to discard or delay packets -- which can harm certain flows and thus lead to suboptimal perceived experience.
+In this document, this is termed 'reactive policy'.
 
-The rapid variation of wireless link capacity and/or bandwidth limitations in networks along with interactive applications that demand low latency and high throughput can lead to poor user experience.
-{{uc}} outlines use cases to illustrate the issues.
+A connection represents the communication link between user (client) and network (router) while a connection session represents the QoS and policy applied to flows within that connection.
+A connection session maybe established using control plane signaling between the client and the network (access) router and is out of this document.
+Transport flows in a connection session may consist of multiple streams such as video or audio.
+The requirements in {{metadata-req}} apply to streams or data units like frames within a flow, but not between flows.
+{{Figure-conn-flow}} shows a high level view of connection sessions, flows and QoS/policy proposed in {{metadata-req}} is applied.
+
+~~~~~~~~
++----------+          +-----------------+                              
+|+----+    |          | +-------------+ |          
+|| A1 |--+ |          | | QoS, Policy | |          
+|+-+--+A2| |          | +---+-----+---+ |          +------+
+|  |+-+--+ |          |     |     |     |          |srv-A2|
+|  |  |    |  conn/session  V     |     |          +--+---+
+|  |  |  *************************|**** |             |
+|  |  +------------- flow-x2 -----|-------------------+   +------+
+|  +------------ flow-x1 ---------|-----------------------+srv-A1|
+|        *************************|**** |                 +--+---+
++-Client-1-+          |           |     |                    |
+                      |           |     |                    |
++----------+  conn/session        V     |                    |
+|+----+  ****************************** |                    |
+|| A1 +------------ flow-x3 ---------------------------------+
+|+----+  ****************************** |
++----------+          +-----------------+
+  Client-2                  Router
+
+~~~~~~~~
+{: #Figure-conn-flow title=”E2E transport flows and connection session”}
+
+{{Figure-conn-flow}} shows Client-1 and Client-2 that negotiate  QoS, policy and other aspects like mobility handling, charging applied to flows in that connection session establised to (access) Router.
+Client-1 has flow-x1 and flow-x2 in its connection session while Client-2 has flow-x3.
+The requirements in this document propose on-path media collaboration signals that apply to streams or data units such as media frames within flows like flow-x1/x2/x3 but not between them.
```

Comment:
Are we limiting the scope of the document to media or including other types of traffic as well to extend the requirements for explicit signaling?

Reply:
Interactive media is already described in the draft (req in section 5.1)
The others include: (a) a limited bandwidth due to large number of users, and a need to prioritize one flow over another (section 5.2 ?), (b) information from the network on, e.g., max bandwidth to allow better rate control.
If we document (b) then we should also refer to sconepro so that it is clear.
The model described in lines 150 - 152 is independent of these requirements pers se.
Do you think we need additional text about all this in the intro section?

Reply:
I was zooming out a bit more to ask if we should include bulk transfer traffic as well to the mix (like backup/restore, downloads/uploads, print jobs etc) so that the document is more generic and not just confined to media traffic. That way, we got more requirements and use-cases as well to bolster the claim that explicit signaling is required to make things more efficient. There are cases where bulk traffic is also in the same application as interactive traffic (like saving a PPT while continuing to type within a remote session (or cloud)). Will that be useful to include these as well in the draft?

Reply:
Agreed would be useful to add.  To keep applicable to the web browser crowd, let's include 'downloading a file' as an example of a lower-priority thing.  With QUIC, that could occur with the same web server as other browsing but is obviously less important than the interactive stuff the user is doing with the same web server.

Reply:
Agree, the use case would be good to document.
And the abstraction in figure with 2 flows (x1/x2)  in a connection session illustrate what the router sees.
Perhaps we should also think if additional requirements for metadata are needed. (my earlier response assumes that the h2n network requirements capture this.)

Reply:
John, are we trying to handle two flows owned by the same application?  Like, I am visiting www.example.com and click on a file to download from files.example.com (a different server)??  

Perhaps we can allow inter-flow priority if a flow is declaring itself as *lower* priority, but not *higher* priority??

I worry that we don't now, or won't in the future, have such a binary "lower priority", though...

Reply:
In the figure, it is 2 flows owned by different servers (srv-1, srv-2) as in the "example.com" example with file download from another server.  The figure was trying to illustrate the GBR/non-GBR (best effort) sessions in 5G, and be general enough to cover the different cases. 

In a way, inter-flow priority of sorts is there in the network, but not signaled.
For example, the problem in Sconepro is that one flow (YouTube) out of many of a non-GBR/ BE session(i.e., has the same signaled connection info between UE - 5G network) is a heavy consumer of bandwidth while other flows in the conn session are not. The flows are all to different servers. The traffic shaper in 5G (UPF) only reduces rate for the high bw user (Youtube) and leaves the others unchanged. My understanding is that Sconepro wants to provide a direct signal as to how the network is constraining that high bw user, i.e., max flow bw.